File: InvoiceService/app/invoice.ts
    export class Invoice {
      Id: string;
      InvoiceNumber: string;
	  InvoiceDate: string;
	  Status: string;
	  Client: string;
	  Total: number;
	  VatTotal: number;
	  Transactions: Transaction[];
    }
	
	export class Transaction {
        Id: string;
        InvoiceId: string;
		Description: string;
		UnitPrice: number;
		Quantity: number;
		Vat: number;
		Total: number;
	}

File: InvoiceService/Models/Invoice.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace InvoiceService.Models
{
    public class Invoice
    {
        public Guid Id { get; set; }
        public string InvoiceNumber { get; set; }
        public string Client { get; set; }
        
        public DateTime InvoiceDate { get; set; }
        public string Status { get; set; }
        public IList<Transaction> Transactions { get; set; }
        public decimal Total { get; set; }
        public decimal VatTotal { get; set; }
    }
}